src/cpp/desktop/DesktopMain.cpp

```diff
@@ -376,6 +376,32 @@ QString inferDefaultRenderingEngine()
 
 QString inferDefaultRenderingEngine()
 {
+   // disable opengl when using nouveau drivers, as a large number
+   // of users have reported crashes when attempting to do so.
+   //
+   // NOTE: we'll currently assume this is fixed in the next Qt
+   // update, so guard only for older Qt for now
+   //
+   // https://github.com/rstudio/rstudio/issues/3781
+   // https://bugreports.qt.io/browse/QTBUG-73715
+#if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)
+   core::system::ProcessResult result;
+   Error error = core::system::runCommand(
+            "lspci -mkv | grep -q 'Driver:[[:space:]]*nouveau'",
```

Comment:
Does `inferDefaultRenderingEngine` only get called on Linux?

Reply:
Each platform gets its own version of this routine (for the three desktop platforms we support). I should probably make this more clear, though.